app/templating/view_context.py

```diff
@@ -85,10 +86,22 @@ def build_view_context_for_question(current_location, variables, rendered_block,
     return context
 
 
-def build_view_context_for_summary(metadata, answer_store, schema_context, variables, csrf_token):
-    rendered_schema_json = renderer.render(g.schema.json, **schema_context)
-    summary_rendered_context = build_summary_rendering_context(g.schema, rendered_schema_json['sections'],
-                                                               answer_store, metadata)
+def build_view_context_for_summary(metadata, answer_store, schema_context, block, variables, csrf_token,
+                                   current_location):
+
+    sections = renderer.render(g.schema.json, **schema_context)['sections']
+    summary_rendered_context = None
+
+    if block == 'Summary':
+        summary_rendered_context = build_summary_rendering_context(g.schema, sections,
+                                                                   answer_store, metadata)
+    elif block == 'SectionSummary':
+        for section in sections:
+            group_ids = {group['id'] for group in section['groups']}
+            if current_location.group_id in group_ids:
+                section = [section]
+                summary_rendered_context = build_summary_rendering_context(g.schema, section, answer_store,
```

Comment:
After this line you want to stop it from looping around the rest of the sections as you have already found your match